### PR TITLE
Fixed #62: Rancher agent connection retry handling

### LINKED ISSUE #62: Agent timeout after 5 minutes not ending gracefully
An agent waits for 5 minutes for server to respond. If at the end of 5 minutes the agent hasn't verified the server is ready, it moves forward as if everything is ok. It needs to do a final check and die gracefully with an error message that it could not reach the server.

### PR DESCRIPTION
**Pull request by cloudnautique**
Added a sleep to the for loop, and have it waiting 10 minutes for the
server to come available. Also, added some additional logging to let
users know it is attempting to reach server, unable to reach or can
reach.

### CODE REVIEW

File: agent/run.sh [hunk 1]
```diff
@@ -235,10 +235,17 @@ upgrade()
 
 wait_for()
 {
+    info "Atempting to connect to: ${CATTLE_URL}"
```
Comment (ibuildthecloud) [inline agent/run.sh]: Typo "Attempting"

File: agent/run.sh [hunk 2]
```diff
@@ -235,10 +235,17 @@ upgrade()
...
     for ((i=0; i < 300; i++)); do
         if ! curl -f -s ${CATTLE_URL} >/dev/null 2>&1; then
             error "${CATTLE_URL}" is not accessible
+            sleep 2
+            if [ "$i" -eq "299" ]; then
+                error "Could not reach ${CATTLE_URL}. Giving up."
+                exit 1
+            fi
         else
+            info "${CATTLE_URL} is accessible"
```
Comment (ibuildthecloud) [inline agent/run.sh]: Why doing something else it dawned on me that echoing CATTLE_URL will cause a problem for support.  When authentication is turned on the CATTLE_URL has a registration token embedded in it which is private to that account.  We don't want to have that URL in logs.  You could sanitize the CATTLE_URL.  So the registration token will look like `http://localhost:8080/v1/scripts/81AB22FBCF352E4498E2:1422684000000:HuMjQsZJPxiBsGYpFo0FrNVv2I`.  So basically `sed 's!/v1/scripts.*!/v1!`

**Comment by cloudnautique** [comment]
Sanitize the url in the method. I did a quick scan, and it wasn't immediately obvious we are dumping the CATTLE_URL var to a log file in other places.

**Comment by ibuildthecloud** [comment]
LGTM

**Pull request merged**